src/components/app/ProfileSharing.js

```diff
@@ -58,11 +62,15 @@ const UploadingStatus = ({ progress }) =>
   </div>;
 
 UploadingStatus.propTypes = {
-  progress: PropTypes.number.isRequired,
+  progress: Number,
 };
 
 class ProfileSharingCompositeButton extends PureComponent {
-  constructor(props) {
+  _permalinkButton: Object;
+  _uploadErrorButton: Object;
+  _permalinkTextField: Object;
```

Comment:
Answering

>  3) I could not find these type definitions. Should I create them or do they exist somewhere already?

`HTMLInputElement` [is already defined by flow](https://github.com/facebook/flow/blob/8b11ef3584b9e4f8b595ba9d41768ccdbf32369e/lib/dom.js#L2738) so you don't need to import it.

As for `ButtonPanel` I made a mistake: I meant `ButtonWithPanel` and not `ButtonPanel`. Because they hold references to the component (see https://github.com/DanielNill/perf.html/blob/4a7590769f01a50d175d3441f32743cf5fc9f178/src/components/app/ProfileSharing.js#L232 where it's defined)